src/workloads/execution/SetWindowFieldsUnbounded.yml

```diff
@@ -0,0 +1,184 @@
+
+  We can divide this up into 4 cases:
+    1. Nothing spills.
+    2. Only the $sort spills.
+    3. Only the $_internalSetWindowFields spills.
+    4. Both the $sort and the $_internalSetWindowFields spill.
+
+  - To control whether the $sort spills, we can use an index, or not.
+  - To control whether the $_internalSetWindowFields spills, we can change the cardinality of
+    the partition key.
+
+Actors:
+- Name: InsertData
+  Type: Loader
+  Threads: 1
+  Phases:
+  - Repeat: 1
+    Database: &db test
+    Threads: 1
+    CollectionCount: 1
+    # Through trial and error, it seems 60k documents, when divided into 12 partitions, does not spill,
+    # but divided into fewer partitions (10 partitions), does spill.
+    DocumentCount: 60000
+    BatchSize: &batchSize 1000
+    Document:
+      # Each document has a unique _id. In phase 2 we use this to create evenly distributed
+      # partition keys.
+      _id: {^Inc: {}}
+      # This int field is what we sum. Its value shouldn't matter for this test.
+      int: &int {^RandomInt: {min: 1, max: 10000}}
+      # This string pads the doc size out to around 10KB.
+      string: &string {^RandomString: {length: 10000}}
+  - Nop: true
+  - Nop: true
+
+- Name: FixupDataAndCreateIndexes
+  Type: RunCommand
+  Threads: 1
+  Phases:
+  - Nop: true
+  - Repeat: 1
+    Database: *db
+    Operations:
+    - OperationName: RunCommand
+      OperationCommand:
+        update: Collection0
+        updates: [
+          {
+            q: {},
+            u: [
+              {$set: {
+                ten_indexed: {$mod: ["$_id", 10]},
+                ten_unindexed: {$mod: ["$_id", 10]},
+                twelve_indexed: {$mod: ["$_id", 12]},
+                twelve_unindexed: {$mod: ["$_id", 12]},
+              }},
+            ],
+            multi: true,
+            upsert: false,
+          }
+        ]
+    - OperationName: RunCommand
+      OperationCommand:
+        createIndexes: Collection0
+        indexes: [
+          {key: {ten_indexed: 1}, name: "ten_indexed_1"},
+          {key: {twelve_indexed: 1}, name: "twelve_indexed_1"},
+        ]
+  - Nop: true
+
+- Name: SetWindowFieldsUnbounded
+  Type: RunCommand
+  Threads: 1
+  Phases:
+  - Nop: true
+  - Nop: true
+  - Repeat: 10
+    Database: *db
+    Operations:
+
+    # Nothing spills:
+    # - $sort is indexed
+    # - partitions are small enough to fit in memory
+    - OperationMetricsName: NothingSpills
+      OperationName: RunCommand
+      OperationCommand:
+        aggregate: Collection0
+        pipeline: [
+          {$setWindowFields: {
+            partitionBy: "$twelve_indexed",
+            output: {
+              total: {$sum: "$int"}
+            }
+          }},
+          {$project: {
+            partitionKey: 1,
+            percentage: {$divide: ["$int", "$total"]},
+            keepString: {$type: "$string"},
+          }}
```

Comment:
I noticed you follow each $setWindowFields with a $project in these benchmarks. Any particular reason?

Reply:
I think the main reason is to avoid sending those 10KB strings in the query response, since the benchmark is trying to focus on the cost of the query rather than the network.  We can't do `{$project: {string: 0}}` though, because it would get pushed down too early in the query, so the documents would all be small, so nothing would spill.

I think the reason for the $divide was just because users would probably do that in queries like this.  I wouldn't expect that to affect the performance much; maybe it's clearer to remove it.

Reply:
Ah I forgot about the 10KB string. That makes sense. Would you mind adding another a comment to that effect?

Reply:
Done